cli/src/cli_util.rs

```diff
@@ -429,12 +429,13 @@ impl CommandHelper {
             Ok(op) => {
                 let repo = workspace.repo_loader().load_at(&op)?;
                 let mut workspace_command = self.for_workable_repo(ui, workspace, repo)?;
-                workspace_command.maybe_snapshot(ui)?;
+                let update_stale = false;
+                workspace_command.maybe_snapshot_and_update_stale(ui, update_stale)?;
 
                 let wc_commit_id = workspace_command.get_wc_commit_id().unwrap();
-                Ok(StaleWorkingCopy::Snapshotted(
-                    workspace_command.repo().store().get_commit(wc_commit_id)?,
-                ))
+                let repo = workspace_command.repo().clone();
+                let commit = repo.store().get_commit(wc_commit_id)?;
+                Ok(StaleWorkingCopy::Snapshotted((repo, commit)))
```

Comment:
Perhaps, this function can return `WorkspaceCommandHelper` instead of `(repo, wc_commit)` pair?

For reference, `for_stale_working_copy()` was a `CommandHelper` method before. 4d0f1b974600fd9c1425b68b5400d51e94515939